File: php/editar.php
<?php 
  include("restclient.php"); 
  $api = new RestClient(array(
     'base_url' => "localhost:8080/api/")
  );

  if(isset($_POST['id']))
  {
    echo "metodo PUT editar.php ";
    $result = $api->put("usuarios/".$_GET['id'], 
		array(
		'email' => $_POST['email'],
		'nombres' => $_POST['nombres'],
		'apellidos' => $_POST['apellidos']
		)
	      );
    echo json_decode($result->response)->message;
  }

  $result = $api->get("usuarios/".$_GET['id']);
  $json_data = json_decode($result->response);

 ?>

<link rel="stylesheet" type="text/css" href="./static/css/bootstrap.min.css">

  <a href="index.php">Indice</a>
  <form method="post" name="formActualizar" class="well" />
    <input type="hidden" id="id" name="id" value="<?php $json_data->_id ?>">
    <div class="row">
      <label class="col-md-3">Email</label>
      <input class="col-md-8" type="text" name="email" value="<?php echo $json_data->email?>" />
    </div>
    
    <div class="row">
      <label class="col-md-3">Nombres</label>
      <input class="col-md-8" type="text" name="nombres" value="<?php echo $json_data->nombres?>" />
    </div>
    
    <div class="row">
      <label class="col-md-3">Apellidos</label>
      <input class="col-md-8" type="text" name="apellidos" value="<?php echo $json_data->apellidos?>" />
    </div>
    
    <button type="submit" class="btn btn-primary">Actualizar usuario</button>
  </form>

File: perl/editar.pl
#!/usr/bin/perl
use LWP::Simple;
use HTTP::Request::Common;
use JSON::PP;
use CGI;
my $query = new CGI;
my $user_id = $query->param('user_id');

my $url = "http://localhost:8080/api/usuarios/".$user_id;

my $mensaje = "";
if ($query->param('email')){
    my $ua = LWP::UserAgent->new;
    my $req = POST($url, ,Content => [
       'email' => $query->param('email'),
       'nombres' => $query->param('nombres'),
       'apellidos' => $query->param('apellidos')
      ]);
    $req->method("PUT");
    
    my $resp = $ua->request($req);
    
    $mensaje = $resp->content;

    # my $post_data = '{ "email": "$query->param(\'email\')", "nombres": "'.$query->param('nombres').'"}';

}


$usuario= decode_json(get($url));

print "Content-type: text/html\n
<html>
  <head>
    <title>Editar usuario</title>
    <link rel='stylesheet' type='text/css' href='/static/css/bootstrap.min.css'>
  </head>
  <body>
    <div class='alert alert-info'>$mensaje</div>
    <form method='POST'>
        <input type='hidden' name='user_id' value='$user_id'>
      <table class='table table-bordered'>
        <th>Email</th>
        <th>Nombres</th>
        <th>Apellidos</th>
        <tr>
          <td> <input type='text' value='$usuario->{email}' name='email' required=required/> </td>
          <td> <input type='text' value='$usuario->{nombres}' name='nombres'/> </td>
          <td> <input type='text' value='$usuario->{apellidos}' name='apellidos'/> </td>
        </tr>
      </table>
      <a href='/' class='btn btn-default'>Volver a listado</a>

      <button type='submit' class='btn btn-primary'>Actualizar Usuario</button> 
    </form>
  </body>
</html>";
